src/MissionManager/VisualMissionItem.cc

```diff
@@ -75,7 +75,12 @@ const VisualMissionItem& VisualMissionItem::operator=(const VisualMissionItem& o
 }
 
 VisualMissionItem::~VisualMissionItem()
-{    
+{
+    if (_currentTerrainAtCoordinateQuery) {
+        disconnect(_currentTerrainAtCoordinateQuery, &TerrainAtCoordinateQuery::terrainDataReceived,
```

Comment:
I though Qt did this automatically when a QObject was deleted? Any signals in/out will be disconnected.